### PR TITLE
Add checkState util.

### PR DESCRIPTION
**Pull request by wayilau**
Motivation:
for wraping illegalStateException throw, link checkNotNull in ObjectUtil.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by wayilau** [comment]
May be i should update other places where throw illegalStateException. @normanmaurer 
Or i will create another PR to do this ?

**Comment by normanmaurer** [comment]
Update others within this PR would be the best

**Comment by wayilau** [comment]
@normanmaurer  
I've update the PR.
But i found that the codestyle is not correct. I have using the codestyle file downloaded from our websit. Could you help me about this ?

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by wayilau** [comment]
CodeStyle.xml is not correct, or i missed somethings ? @normanmaurer

File: codec-http/src/main/java/io/netty/handler/codec/http/CombinedHttpHeaders.java [hunk 1]
```diff
@@ -92,9 +93,7 @@ public CharSequence escape(CharSequence value) {
                 return itr;
             }
             Iterator<CharSequence> unescapedItr = unescapeCsvFields(itr.next()).iterator();
-            if (itr.hasNext()) {
-                throw new IllegalStateException("CombinedHttpHeaders should only have one value");
-            }
+            checkState(!itr.hasNext(), "CombinedHttpHeaders should only have one value");
```
Comment (johnou) [inline codec-http/src/main/java/io/netty/handler/codec/http/CombinedHttpHeaders.java]: Tbh checkState seems ambiguous, assertTrue? assertFalse?

Comment (wayilau) [reply codec-http/src/main/java/io/netty/handler/codec/http/CombinedHttpHeaders.java]: i  will check, do you have a correct codestyle.xml ? @johnou

Comment (johnou) [reply codec-http/src/main/java/io/netty/handler/codec/http/CombinedHttpHeaders.java]: https://github.com/netty/netty-build/blob/master/src/main/resources/io/netty/checkstyle.xml

Comment (wayilau) [reply codec-http/src/main/java/io/netty/handler/codec/http/CombinedHttpHeaders.java]: Not this one, i said it's codeStyle.xml, i downloaded from netty-website,imported to idea IDE. But when i reformat code,  i found it's not correct. I confused.

**Pull request closed without merging**